app/models/energy/ScheduleRuleset.py

```diff
@@ -0,0 +1,207 @@
+"""Schedule Ruleset Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+
+
+class DateSpecificationType(str, Enum):
+
+    date_range = 'DateRange'
+    specific_date = 'SpecificDate'
+
+
+class ScheduleContinuous(BaseModel):
+    """This Numeric Type allows all numbers, including fractional amounts, within the range to
+  be valid."""
+
+    schedule_continuous: float = Schema(
+        ...,
+    )
+
+
+class ScheduleDiscrete(BaseModel):
+    """This Numeric Type allows all only integers within the range to be valid."""
+
+    schedule_discrete: int = Schema(
+        ...,
+    )
+
+
+class NumericType (BaseModel):
+    """Designates how the range values are validated."""
+
+    numeric_type:  Union[ScheduleContinuous, ScheduleDiscrete]
+
+
+class UnitType (str, Enum):
```

Comment:
Should this class be repeated for both the `ShceduleFile` and the `ScheduleRuleset`?  I feel that there should be some way to have these two reference the same object.

Reply:
I have made changes, so that `UnitType` is being referenced in `ScheduleFile`.

Reply:
Common way to do this is factor shared content into a third file, maybe `ScheduleBase` or `ScheduleCommon`